src/aks-preview/azext_aks_preview/decorator.py

```diff
@@ -135,6 +141,59 @@ def init_nat_gateway_models(self) -> None:
         # for model_name, model_type in nat_gateway_models.items():
         #     setattr(self, model_name, model_type)
 
+    def _get_pod_identity_models(self) -> Dict:
+        """Get a dictionary of pod identity related models.
+
+        The models are stored in a dictionary, the key is the model name and the value is the model type.
+
+        :return: Dict
+        """
+        pod_identity_models = {}
+        pod_identity_models["ManagedClusterPodIdentityProfile"] = self.__cmd.get_models(
+            "ManagedClusterPodIdentityProfile",
+            resource_type=self.resource_type,
+            operation_group="managed_clusters",
+        )
+        pod_identity_models["ManagedClusterPodIdentityException"] = self.__cmd.get_models(
+            "ManagedClusterPodIdentityException",
+            resource_type=self.resource_type,
+            operation_group="managed_clusters",
+        )
+        return pod_identity_models
+
+    @property
+    def pod_identity_models(self) -> SimpleNamespace:
+        """Get pod identity related models.
+
+        The models are stored in a SimpleNamespace object, could be accessed by the dot operator like
+        `pod_identity_models.ManagedClusterPodIdentityProfile`.
+
+        :return: SimpleNamespace
+        """
+        if self.__pod_identity_models is None:
+            self.__pod_identity_models = SimpleNamespace(**self._get_pod_identity_models())
+        return self.__pod_identity_models
+
+    def flattern_to_class_attributes(self, models: Union[Dict, SimpleNamespace]) -> None:
```

Comment:
question: what is the function for?

Reply:
In case we want to get the model directly from an instance of `AKSPreviewModel`. But this is not used anywhere currently.

Reply:
Can we remove this if it's not being used anywhere? Why we want to copy a field to the class? IMO, it would cause more confuse to other developers, especially under a dynamic language context.

Reply:
Updated (removed) in commit [#91c0ca3](https://github.com/Azure/azure-cli-extensions/pull/4243/commits/91c0ca36c0f84ef93033e1de3e98cf5596dfb32b).